### PR TITLE
feat: add error, suggestion, and explanation as inline PR comments

### PR DESCRIPTION
**Pull request by anuraagnalluri**

### CODE REVIEW

File: internal/suggestions/suggestion.go [hunk 1]
```diff
@@ -22,42 +31,81 @@ func Suggest(docPath, maxSuggestions string) (string, error) {
...
 }
 
-func formatSuggestionsAndExplanations(body prBodyInfo) string {
-	var output string
-	for i := 0; i < len(body.suggestions); i++ {
-		output += fmt.Sprintf("**Suggestion %d**: %s\n\n", i+1, body.suggestions[i])
-		output += fmt.Sprintf("**Explanation %d**: %s\n\n", i+1, body.explanations[i])
+func formatComment(err, suggestion, explanation string, index int) string {
+	var commentParts []string
+	commentParts = append(commentParts, fmt.Sprintf("**Error %d**: %s", index, err))
```
Comment (chase-crumbaugh) [inline internal/suggestions/suggestion.go:57]: Nice, I was noticing that it was a bit difficult to understand the PR bodies without the actual errors

File: internal/suggestions/suggestion.go [hunk 2]
```diff
@@ -22,42 +31,81 @@ func Suggest(docPath, maxSuggestions string) (string, error) {
...
+	commentParts = append(commentParts, fmt.Sprintf("**Suggestion %d**: %s", index, suggestion))
+	commentParts = append(commentParts, fmt.Sprintf("**Explanation %d**: %s", index, explanation))
+	return strings.Join(commentParts, "\n\n")
+}
+
+func formatBody(info prCommentsInfo) string {
+	var bodyParts []string
+	for i := 0; i < len(info.lineNums); i++ {
+		if info.lineNums[i] == 0 {
+			bodyParts = append(bodyParts, formatComment(info.errs[i], info.suggestions[i], info.explanations[i], i+1))
```
Comment (chase-crumbaugh) [inline internal/suggestions/suggestion.go:67]: Given that we are adding all the inline comments, I'm wondering if we can make the PR bodies a bit more succinct. For example, a bulleted list of just the suggestions (no error or explanation) would basically serve as a list of the changes contained in the PR

Comment (anuraagnalluri) [reply internal/suggestions/suggestion.go]: Bodies will be pretty succinct. Currently, I don't want to fail the action if it's unable to leave a comment (for whatever reason), so we use the body as a fallback to catch the logging of those errors, suggestions, and explanations if and when that occurs.

File: internal/git/git.go [hunk 3]
```diff
@@ -27,7 +27,7 @@ import (
 	"github.com/speakeasy-api/sdk-generation-action/internal/logging"
 	"github.com/speakeasy-api/sdk-generation-action/pkg/releases"
 
-	"github.com/google/go-github/v48/github"
+	"github.com/google/go-github/v54/github"
```
Comment (chase-crumbaugh) [inline internal/git/git.go:30]: Do we need to confirm that all our other requests still work? I can't imagine they wouldn't but might be wise to just run through a quick SDK setup in `dev`

File: internal/git/git.go [hunk 4]
```diff
@@ -472,6 +472,25 @@ func (g *Git) WritePRBody(prNumber *int, body string) error {
 	return nil
 }
 
+func (g *Git) WritePRComment(prNumber *int, fileName, body string, line int) error {
```
Comment (chase-crumbaugh) [inline internal/git/git.go:475]: nit: does prNumber need to be a pointer? Is it an option to leave it as nil?

Comment (chase-crumbaugh) [reply internal/git/git.go:475]: Also, should this method be in the github client? `github.go`? Seems like more of a github thing than a git thing

File: internal/suggestions/suggestion.go [hunk 5]
```diff
@@ -5,12 +5,23 @@ import (
...
+	"github.com/speakeasy-api/sdk-generation-action/internal/logging"
+	"regexp"
+	"strconv"
 	"strings"
 )
 
-type prBodyInfo struct {
+var (
+	fileNameRegex      = regexp.MustCompile(`^(.*?(\.yaml|\.yml|\.json))`)
+	outputFileRegex    = regexp.MustCompile(`Suggestions applied and written to (.+)`)
```
Comment (chase-crumbaugh) [inline internal/suggestions/suggestion.go:16]: I think this line is only written if parallelization (or autonext or whatever) is enabled? Is there a weird case where we accidentally change that and break this?

File: internal/git/git.go (hunk 4 above)
Comment (anuraagnalluri) [reply internal/git/git.go]: Changed signature to `int`.

Regarding github vs. git, we don't have that separation elsewhere in the action. All the logic for creating/updating gen PRs (which uses `go-git`) is muddled with logic that gets the latest tag or download link (which uses `go-github`). The `Git` struct in `git.go` has clients for both of these. Can separate it out but would rather refactor it in a different PR.

File: internal/suggestions/suggestion.go (hunk 5 above)
Comment (anuraagnalluri) [reply internal/suggestions/suggestion.go:16]: Works in `--serial`, but yes this line isn't written when auto-continue is disabled. That should be fine as this is action code, and we'd always run the CLI with it enabled in this context.

**Review by chase-crumbaugh — APPROVED** [review]

File: internal/git/git.go (hunk 4 above)
Comment (chase-crumbaugh) [reply internal/git/git.go]: Gotcha 👍

File: internal/git/git.go (hunk 3 above)
Comment (anuraagnalluri) [reply internal/git/git.go:30]: works fine: https://github.com/speakeasy-sdks-dev/test-petstore

**Pull request merged**